lfs/transfer_queue.go

```diff
@@ -28,6 +28,68 @@ type Transferable interface {
 	LegacyCheck() (*api.ObjectResource, error)
 }
 
+type retryCounter struct {
+	// MaxRetries is the maximum number of retries a single object can
+	// attempt to make before it will be dropped.
+	MaxRetries int `git:"lfs.transfer.maxretries"`
+
+	// cmu guards count
+	cmu sync.Mutex
+	// count maps OIDs to number of retry attempts
+	count map[string]int
+}
+
+// newRetryCounter instantiates a new *retryCounter. It parses the gitconfig
+// value: `lfs.transfer.maxretries`, and falls back to defaultMaxRetries if none
+// was provided.
+//
+// If it encountered an error in Unmarshaling the *config.Configuration, it will
+// be returned, otherwise nil.
+func newRetryCounter(cfg *config.Configuration) *retryCounter {
+	rc := &retryCounter{
+		MaxRetries: defaultMaxRetries,
+
+		count: make(map[string]int),
+	}
+
+	if err := cfg.Unmarshal(rc); err != nil {
+		tracerx.Printf("rc: error parsing config, falling back to default values...: %v", err)
```

Comment:
This should never fail. If the value stored in the `.gitconfig` isn't parsable as an `int`, we'll just [fall back to the default](https://github.com/github/git-lfs/blob/06b4c1463df60e4c480ff6d9c14fd81fcada3c17/config/config.go#L165), the current value (which happens to be `defaultMaxRetries`).